### PR TITLE
Feature/validation tests pure

### PR DESCRIPTION
**Pull request by grimadas**
## What is this pull request?
Fixed stateful validation with tests. 
   
## Why do you implement it? Why do we need this pull request?
Now for each stateful validation there is a test proving work of the validation.
For the test we use Mock object hence there is no dependency from other modules.  
  
## How to use the features provided in the pull request?
For each command there are tests, that you can run.  
Each command has tests for valid and not valid cases. 
You are welcome to add new/missed cases for commands.

### CODE REVIEW

File: irohad/model/impl/stateful_command_validation.cpp [hunk 1]
```diff
@@ -123,7 +123,7 @@ namespace iroha {
       // Creator must have permission to create assets
       return creator.permissions.create_assets &&
              // Name is within some range
-             asset_name.size() > 0 && asset_name.size() < 8 &&
+             asset_name.size() > 0 && asset_name.size() < 10 &&
```
Comment (l4l) [inline irohad/model/impl/stateful_command_validation.cpp]: wtf is 10?

Comment (grimadas) [inline irohad/model/impl/stateful_command_validation.cpp]: Just some number, can be changed any time

Comment (grimadas) [inline irohad/model/impl/stateful_command_validation.cpp]: Max length of asset name

File: irohad/model/impl/stateful_command_validation.cpp [hunk 2]
```diff
@@ -211,13 +212,17 @@ namespace iroha {
...
           // Can account transfer assets
           creator.permissions.can_transfer &&
           // Creator can transfer only from their account
           creator.account_id == src_account_id &&
           // Balance in your wallet should be at least amount of transfer
           account_asset.value().balance >=
-              amount.get_joint_amount(asset.value().precision);
+              amount.get_joint_amount(asset.value().precision) &&
+          // Amount must be not zero
+          (amount.frac_part > 0 || amount.int_part > 0);
```
Comment (MizukiSonoko) [inline irohad/model/impl/stateful_command_validation.cpp]: Asset's amount will not be zero? why?

Comment (grimadas) [inline irohad/model/impl/stateful_command_validation.cpp]: If you allow transfer zero assets - you ddos the system

Comment (MizukiSonoko) [inline irohad/model/impl/stateful_command_validation.cpp]: ok i see

**Pull request merged**